android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java

```diff
@@ -2259,16 +2259,11 @@ public void setBufferConfig(BufferConfig config) {
         initializePlayer();
     }
 
-    public void setDrmType(UUID drmType) {
-        this.drmUUID = drmType;
-    }
-
-    public void setDrmLicenseUrl(String licenseUrl){
-        this.drmLicenseUrl = licenseUrl;
-    }
-
-    public void setDrmLicenseHeader(String[] header){
-        this.drmLicenseHeader = header;
+    public void setDrm(DRMProps drmProps) {
+        this.drmProps = drmProps;
+        if (drmProps != null && drmProps.getDrmType() != null) {
+            this.drmProps.setDrmUUID(Util.getDrmUuid(drmProps.getDrmType()));
+        }
```

Comment:
```java
        if (drmProps != null && drmProps.getDrmType() != null) {
            this.drmProps.setDrmUUID(Util.getDrmUuid(drmProps.getDrmType()));
        }
```
I think above code could be moved inside the DrmProps.parse fn.

Reply:
I understand. but for that reason, it seems like `parse` function should be categorized in a different class. (for example, `DRMUtils`) to be precise, the problem seems to be that `DRMProp` doesn't just act as a Data class.

If you worried deep dependencies, I recommend copying [media3's implementation](https://github.com/androidx/media/blob/1.1.0/libraries/common/src/main/java/androidx/media3/common/util/Util.java#L2067-L2083).

However, I am not suggesting it because it's a problem, so you don't have to fix it.👍